### PR TITLE
Rentify 13 create favourites

### PR DESCRIPTION
**Pull request by Slavchev03**

### CODE REVIEW

File: back-end/src/main/java/com/devminds/rentify/entity/Item.java [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package com.devminds.rentify.entity;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import jakarta.persistence.*;
```
Comment (maria-markova) [inline back-end/src/main/java/com/devminds/rentify/entity/Item.java:5]: Pls remove wildcard import

File: back-end/src/main/java/com/devminds/rentify/entity/Picture.java [hunk 2]
```diff
@@ -1,12 +1,6 @@
 package com.devminds.rentify.entity;
 
-import jakarta.persistence.Column;
-import jakarta.persistence.Entity;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.GenerationType;
-import jakarta.persistence.Id;
-import jakarta.persistence.ManyToOne;
-import jakarta.persistence.Table;
+import jakarta.persistence.*;
```
Comment (maria-markova) [inline back-end/src/main/java/com/devminds/rentify/entity/Picture.java:3]: Remove wildcard imports.

File: front-end/src/App.jsx [hunk 3]
```diff
@@ -9,6 +9,8 @@ import ItemDetails from "./components/items/ItemDetails";
 import Login from "./components/login/Login";
 import CreateItem from "./components/add-item/Create-item";
 import Register from "./components/login/Register";
+// import LikedItemsPage from "./components/items/LikedItemsPage";
```
Comment (maria-markova) [inline front-end/src/App.jsx:12]: remove comment

File: back-end/src/main/java/com/devminds/rentify/config/AuthConfig.java [hunk 4]
```diff
@@ -37,9 +37,9 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 .requestMatchers("/rentify/items/**").permitAll()
                                 .requestMatchers("/rentify/addresses").permitAll()
                                 .requestMatchers("/rentify/addresses/**").permitAll()
-                                .requestMatchers("/rentify/images").permitAll()
-                                .requestMatchers("/rentify/images/**").permitAll()
                                 .requestMatchers("/rentify/favorite/**").permitAll()
+                                .requestMatchers("/rentify/favorite/userFavorite/**").permitAll()
```
Comment (ValentinTsonevskii) [inline back-end/src/main/java/com/devminds/rentify/config/AuthConfig.java:41]: typo in favourites

File: back-end/src/main/java/com/devminds/rentify/controller/LikeController.java [hunk 5]
```diff
@@ -15,6 +14,8 @@
 import org.springframework.web.bind.annotation.*;
 
 import java.util.Optional;
+import java.util.Set;
+
 
 @RestController
 @RequestMapping("/rentify/favorite")
```
Comment (ValentinTsonevskii) [inline back-end/src/main/java/com/devminds/rentify/controller/LikeController.java:21]: typo in favourites

File: back-end/src/main/java/com/devminds/rentify/controller/LikeController.java [hunk 6]
```diff
@@ -49,4 +55,14 @@ public ResponseEntity<String> likeItem(@RequestBody LikeDto likeDto) {
     }
 
 
+    @GetMapping("/userFavorite/{userId}")
```
Comment (ValentinTsonevskii) [inline back-end/src/main/java/com/devminds/rentify/controller/LikeController.java:58]: typo in favourites

File: back-end/src/main/java/com/devminds/rentify/entity/Item.java [hunk 7]
```diff
@@ -62,14 +66,14 @@ public class Item {
 
     @ManyToOne
     private Address address;
+//
+//    @OneToMany
+//    private List<History> histories;
```
Comment (ValentinTsonevskii) [inline back-end/src/main/java/com/devminds/rentify/entity/Item.java]: remove comment or uncomment it

File: front-end/src/components/items/ItemsList.jsx [hunk 8]
```diff
@@ -94,17 +105,52 @@ const ItemsList = ({ searchTerm }) => {
...
-    // TODO: Add logic for post request to add or remove a like
+    const requestBody = {
+      itemId: itemId,
+      userId: parseInt(userId, 10),
+      isLiked: isLiked,
+    };
+
+    try {
+      const response = await fetch(
+        "http://localhost:8080/rentify/favorite/liked",
```
Comment (ValentinTsonevskii) [inline front-end/src/components/items/ItemsList.jsx:132]: typo in favourites

File: front-end/src/components/items/LikedItemsPage.jsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+          const decoded = jwtDecode(token);
+          const userId = decoded.jti;
+
+          const response = await fetch(`http://localhost:8080/rentify/favorite/userFavorite/${userId}`, {
+            method: 'GET',
+          });
+
+          if (response.ok) {
+            const likedItemsIds = await response.json();
+            // Fetch additional details for each liked item
```
Comment (ValentinTsonevskii) [inline front-end/src/components/items/LikedItemsPage.jsx:26]: remove comment

File: front-end/src/components/items/LikedItemsPage.jsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+        }
+      } catch (error) {
+        console.error('Error fetching liked items:', error.message);
+      }
+    };
+
+    fetchLikedItemsFromDB();
+  }, []);
+
+  console.log("haresanii")
```
Comment (ValentinTsonevskii) [inline front-end/src/components/items/LikedItemsPage.jsx:49]: remove unnesessery console logs

File: back-end/src/main/java/com/devminds/rentify/entity/Picture.java (hunk 2 above)
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/java/com/devminds/rentify/entity/Picture.java:3]: This is still a wildcard import. Change IntelliJ settings:
Settings > Editor > Code Style > Java -> set "Class count..." and "Names count..." to 99

**Review by KalinVasilev-at-blankfactor — APPROVED** [review]

**Review by aleksandramihaylova — APPROVED** [review]

**Pull request merged**